### ISSUE TITLE
Invalid "use strict is unnecessary" warning for ES6 modules

### ISSUE THREAD

**Issue by ChadKillingsworth** [google/closure-compiler#1263]
Chrome requires es6 classes to explicitly be in strict mode when used. However, since ES6 modules are rewritten to goog.module, the following warning is issued:

```
src/file.js:17: WARNING - use strict is unnecessary in goog.module files.
```

This warning is incorrect since 'use strict' is required for uncompiled mode support.

[Example usage](https://github.com/ChadKillingsworth/geolocation-marker/blob/master/src/geolocation-marker.js)

**Comment by concavelenz** [comment]
ES6 and goog.modules are are implicitly strict, there is no need to mark them as "use strict".  If the final result is not strict that is a problem in the rewriting.  The warning is correct.

**Comment by ChadKillingsworth** [comment]
I realize that's what the spec says - but it doesn't match reality. In chrome:

> Uncaught SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode

[Test case](https://jsfiddle.net/snzvjbso/)

Currently you have to choose to either get a compiler warning or not to support uncompiled mode.

**Comment by nicks** [comment]
Ya, Chad is right. V8 refuses to execute es6 without an explicit 'use
strict'. We've run into this too.

**Comment by blickly** [comment]
I think I'm missing something. How are you running ES6 modules uncompiled? I thought it wasn't yet supported in Chrome.

**Comment by nicks** [comment]
classes landed in Chrome 42.

**Comment by MatrixFrog** [comment]
Classes are implemented in Chrome but ES6 modules are not as far as I know. I think part of the confusion is that the error message at https://github.com/google/closure-compiler/blob/4398f4e047971c192178cb07d68da822f2871306/src/com/google/javascript/jscomp/ProcessEs6Modules.java#L56 is referring to ES6 modules not goog.modules. ES6 modules (currently) are rewritten to goog.provide/require, not to goog.module. Just sent out a CL to change the wording of that message.

**Comment by concavelenz** [comment]
@ChadKillingsworth is the issue that the rewritten ES6 modules don't have a "use strict" in the output?

**Comment by ChadKillingsworth** [comment]
@concavelenz No - the issue is that if the file is used as uncompiled source (not a module), Chrome requires "use strict" to be present.

I don't really understand the purpose of the warning. In a standards compliant browser, "use strict" would be unnecessary, but it most certainly isn't causing harm either. At best this seems like a lint waring.

**Comment by concavelenz** [comment]
@ChadKillingsworth You have an ES6 module that has neither import or exports, is that correct? Otherwise how are you using it as a module?

**Comment by ChadKillingsworth** [comment]
Chrome has updated - this is no longer an issue.
